ktxspec.adoc

```diff
@@ -744,14 +744,48 @@ These 3 items combined form a _Data Format Descriptor_
 The full specification for this is found in the _Khronos Data
 Format Specification_ version 1.3 <<KDF13>>.
 
-If the _dfDescriptor_ describes an sRGB transfer function then
-`<<_vkformat,vkFormat>>` must be one of the _SRGB_ formats.
-
 The _dfDescriptor_ is partially expanded here in order to provide
 sufficient information for a KTX2 file to be parsed without having to
 refer to <<KDF13>>. It consists of one or more _Descriptor Blocks_
 (dfDescriptorBlock).
 
+The following restrictions must be obeyed when setting the fields of the
+DFD. The value of `<<_vkformat,vkFormat>>` and the image data must never
+contradict the DFD.
+
+* `color_model` must be `KHR_DF_MODEL_RGBSDA`, `KHR_DF_MODEL_YUVSDA`
+  or one of the block compressed color models listed in <<KDF13>> Section
+  5.6 or its successors, currently `KDF_DF_MODEL_BC1A` to
+  `KDF_DF_MODEL_PVRTC2`.
+* `color_primaries` may be any of the available values since conversion
+  to a display's primaries is relatively straightforward.
```

Comment:
It is "relatively straightforward" only when the white point is the same. Otherwise it's a bit more complicated (from that NVIDIA guide):
> In addition to simply changing the color primaries, it may be necessary to adjust the white point of the image as well. The standard way of doing this is through a Chromatic Adaptation Transform (CAT). The common CAT solutions are the Bradford or Von Kries transforms. These convert the colors to an LMS color space, then scale by destination white / source white (also in LMS space), and finally convert back from LMS space. As the transform to LMS space in both Bradford and Von Kries is simply a 3x3 matrix multiply, the entire CAT can be collapsed into a single 3x3 multiply, just like the transform of primaries.

Do matrices provided in the DFD spec accommodate for this?

Reply:
It boils down to a single matrix multiply, which of course the hardware is very good at doing, that converts both primaries and white point. This single matrix multiply is what I mean by relatively straightforward. Can you propose some alternative language. I certainly don't want to get into details of color space conversion here.

I'll check the matrices in the DFD spec. In any case the color spaces are all standard which proper specifications and ICC profiles which will provide the necessary information.

Reply:
I think the matrices in KDFS spec. are for von Kries transform but, given the nature of the KDFS spec., I think the chapter 14 introduction is explanatory. I do not think that KTX2 needs to specify what CAT should be used. If glTF starts to allow different primaries it may want to mandate a CAT. HTML5 should probably specify the CAT to be used by browsers when converting from sRGB/BT.709 primaries to the display primaries.

I'll change the language to something along the lines of 

> ... conversion of the primaries and white point to those of a display's can be done by a 3x3 matrix multiply.

Reply:
> HTML5 should probably specify the CAT...

If only web standards were so specific...

Rendering of HTML Images with embedded ICC profiles is up to implementation, Canvas API should someday consistently support rendering to float16 surfaces (IIUC, with extended sRGB).

I'm fine with the latest proposed language.